src/compiler/checker.ts

```diff
@@ -2278,12 +2278,19 @@ module ts {
                 }
             }
             else if (links.type === resolvingType) {
-                links.type = anyType;
-                if (compilerOptions.noImplicitAny) {
-                    let diagnostic = (<VariableLikeDeclaration>symbol.valueDeclaration).type ?
-                        Diagnostics._0_implicitly_has_type_any_because_it_is_referenced_directly_or_indirectly_in_its_own_type_annotation :
-                        Diagnostics._0_implicitly_has_type_any_because_it_is_does_not_have_a_type_annotation_and_is_referenced_directly_or_indirectly_in_its_own_initializer;
-                    error(symbol.valueDeclaration, diagnostic, symbolToString(symbol));
+                if ((<VariableLikeDeclaration>symbol.valueDeclaration).type) {
+                    // Variable has type annotation that circularly references the variable itself
+                    links.type = unknownType;

```

Comment:
Why the unknown type here as opposed to the 'any' type?  What's the mental model i should use for when i should use the 'unknown' type?  Thanks!  (also a comment in the code explaining this would be useful).